backend/planty/dashboard/views.py

```diff
@@ -290,3 +295,234 @@ def post(self, request: Request):
         plant.save()
 
         return Response(status=status.HTTP_200_OK)
+
+
+class InstructionsView(APIView):
+    def get(self, request: Request):
+        user: User = request.user
+        my_instructions = Instruction.objects.filter(user=user, public='False')
+
+        my_instructions_json = []
+        for instruction in my_instructions:
+            instruction_json = {
+                'id': str(instruction.id),
+                'name' : instruction.name,
+                'species': instruction.species,
+
+                'watering': instruction.watering,
+                'insolation': instruction.insolation,
+                'fertilizing': instruction.fertilizing,
+            }
+            my_instructions_json.append(instruction_json)
+
+
+        if request.method == 'GET' and 'limit_popular' in request.GET:
+            N = request.GET['limit_popular']
+        else:
+            N = 10
+
+        suggested = Instruction.objects.filter(public='True')
+        # returns best N suggested instruction
+        desc_suggested_instruction = suggested.order_by('num_selected').reverse()[:N]
+
+
+        desc_suggested_instructions_json = []
+        for instruction in desc_suggested_instruction:
+            instruction_json = {
+                'id': str(instruction.id),
+                'name': instruction.name,
+                'species': instruction.species,
+
+                'watering': instruction.watering,
+                'insolation': instruction.insolation,
+                'fertilizing': instruction.fertilizing,
+
+                'num_selected': instruction.num_selected
+            }
+            desc_suggested_instruction.append(instruction_json)
+
+        instructions_json = {
+            my: my_instructions_json,
+            suggested: desc_suggested_instructions_json
+        }
+
+        return Response(instructions_json, status=status.HTTP_200_OK)
+
+    def post(self, request: Request):
+        user: User = request.user
+        serializer = InstructionCreateSerializer(data=request.data)
+
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        data = serializer.validated_data
+
+        instruction: Instruction = Instruction.objects.create(
+            id=uuid4(),
+            name=name,
+
+            user=user,
+            species=data['species'],
+
+            watering=data['watering'],
+            insolation=data['insolation'],
+            fertilizing=data['fertilizing']
+        )
+        instruction.save()
+
+        return Response(status=status.HTTP_201_CREATED)
+
+    def put(self, request: Request):
+        user: User = request.user
```

Comment:
Do PUT i DELETE powinno się przekazywać id przez URL

np. `DELETE /dashboard/instructions/abcd1234`

to będzie chyba wymagało przeniesienia tych dwóch metod do osobnej klasy i dodania osobnego urlpattern `/instructions/<uuid:id>`

w widoku można się do tego dostać chyba przez self.args i self.kwargs

Reply:
Dużo się tutaj nie zmieni, jedynie sposób w jaki przekazujemy id, więc nie wiem czy trzeba tu coś ustalac. Ale jeśli chcesz to można się zgadać.

Reply:
jeśli dobrze zrozumiałem przekazywanie id, to PUT i DELETE gotowe

Reply:
Prawie, z tego co wiem to w class-based views trzeba użyć self.kwargs, a nie przez argumenty metody.

Reply:
Najlepiej będzie mieć kilka klas:
* `InstructionsView` - obsługuje GET i POST. GET dla dostania wszystkich swoich instrukcji, a POST do stworzenia nowej. To byłoby pod url'em `.../instructions/`
* `InstructionView` - pod url'em `.../instructions/<uuid:id>` - GET będzie do dostania danych o konkretnej instrukcji, PUT do edycji, i DELETE do usuwania
* `PopularInstructionsView` jest git moim zdaniem
* `ShareInstructionView` jest też git, ale dałbym ten POST też bez żadnego request body. Natomiast zrobić to pod url'em `.../instructions/<uuid:id>/share/`

Reply:
Czy to zostało zrobione?

> Najlepiej będzie mieć kilka klas:
> 
> * `InstructionsView` - obsługuje GET i POST. GET dla dostania wszystkich swoich instrukcji, a POST do stworzenia nowej. To byłoby pod url'em `.../instructions/`
> * `InstructionView` - pod url'em `.../instructions/<uuid:id>` - GET będzie do dostania danych o konkretnej instrukcji, PUT do edycji, i DELETE do usuwania
> * `PopularInstructionsView` jest git moim zdaniem
> * `ShareInstructionView` jest też git, ale dałbym ten POST też bez żadnego request body. Natomiast zrobić to pod url'em `.../instructions/<uuid:id>/share/`

Reply:
tak